src/dispatcher.cc

```diff
@@ -115,6 +118,37 @@ int Dispatcher::Close(int sockfd) {
   }
 }
 
+int Dispatcher::Getaddrinfo(const char* node, const char* service, const addrinfo* hints, addrinfo** res) {
+  // TODO: Check if service/port is ever used
+  // [pid 108970] client->getaddrinfo("google.de", nil, 0xc00009c090, 0xc0000a4010)                       = 0
+
+  for (auto& el : Conf()["tls"].items()) {
+    std::string domain = el.key().substr(0, el.key().find(':'));
+
+    if (std::string(node) == domain) {
+      // get all IPs
+      int ret = raw_->Getaddrinfo(node, service, hints, res);
+      if (ret != 0) {
+        return ret;
+      }
```

Comment:
Do this at the very beginning of the function. (And replace last line with return 0)